File: be/user_be_java/src/main/java/com/user/controller/SysController.java
package com.user.controller;

import com.user.common.response.ApiResponse;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;
import java.util.Map;

/**
 * 系统控制器
 */
@RestController
@RequestMapping("/api/sys")
@RequiredArgsConstructor
@Tag(name = "Sys", description = "系统接口")
public class SysController {

    @Value("${app.be-type}")
    private String beType;

    /**
     * 获取后端类型
     */
    @GetMapping("/be_type")
    @Operation(summary = "获取后端类型")
    public ResponseEntity<ApiResponse<Map<String, String>>> getBeType() {
        Map<String, String> result = new HashMap<>();
        result.put("beType", beType);
        return ResponseEntity.ok(ApiResponse.success(result));
    }
}

File: be/user_be_go/internal/controller/sys_controller.go
package controller

import (
	"user_be_go/internal/config"
	"user_be_go/pkg/response"

	"github.com/gin-gonic/gin"
)

type SysController struct{}

func NewSysController() *SysController {
	return &SysController{}
}

// GetBeType 获取后端类型
// @Summary 获取后端类型
// @Description 获取当前后端系统类型
// @Tags Sys
// @Accept json
// @Produce json
// @Success 200 {object} response.ApiResponse
// @Router /api/sys/be_type [get]
func (c *SysController) GetBeType(ctx *gin.Context) {
	response.Success(ctx, map[string]string{
		"beType": config.AppConfig.BeType,
	})
}

